src/vs/workbench/browser/parts/quickinput/quickInput.ts

```diff
@@ -530,6 +537,20 @@ class QuickPick<T extends IQuickPickItem> extends QuickInput implements IQuickPi
 								this.ui.list.domFocus();
 							}
 							break;
+						case KeyCode.Enter:
+						case KeyCode.Ctrl:
+						case KeyCode.Shift:
+						case KeyCode.LeftArrow:
```

Comment:
Is there a better way to know if the key will update the undo history? Should Alt be part of this list?

Does this work as expected when the user presses right arrow and then a letter?